R/eval-tidy.R

```diff
@@ -342,3 +339,12 @@ f_self_eval <- function(overscope, overscope_top) {
     .Call(rlang_eval, f_rhs(f), overscope)
   }
 }
+tilde_eval <- function(f, env) {
+  # Inline the base primitive because overscopes override `~` to make
+  # quosures self-evaluate
+  mut_node_car(f, base::`~`)
+
+  # Change both old and new formulas to original head avoid side-effects
+  on.exit(mut_node_car(f, sym_tilde))
```

Comment:
I worry about this implementation as it seem unlikely to interact well with the byte code compiler